### PR TITLE
always use pg_dump for DB backups

### PR DESCRIPTION
**Pull request by evgeni**
- **always use pg_dump for DB backups**
- **drop --include-db-dumps parameter, we always include them now**
- **unify DB backup steps**

includes https://github.com/theforeman/foreman_maintain/pull/891 for the checks

### CODE REVIEW

File: definitions/procedures/backup/offline/candlepin_db.rb [hunk 1]
```diff
@@ -1,53 +0,0 @@
...
-  module Offline
-    class CandlepinDB < ForemanMaintain::Procedure
-      metadata do
-        description 'Backup Candlepin DB offline'
-        tags :backup
-        label :backup_offline_candlepin_db
-        for_feature :candlepin_database
-        preparation_steps { Checks::Candlepin::DBUp.new unless feature(:candlepin_database).local? }
-        param :backup_dir, 'Directory where to backup to', :required => true
-        param :tar_volume_size, 'Size of tar volume (indicates splitting)'
```
Comment (evgeni) [inline definitions/procedures/backup/offline/candlepin_db.rb:11]: It's funny this was defined, but never passed to the procedure (heck, even the CLI param is called `--split-pulp-tar`), so yay less unused complexity?

**Comment by evgeni** [comment]
```
# foreman-maintain backup offline --incremental /var/off-size/katello-backup-2024-06-26-10-03-42/ /var/lol
Starting backup: 2024-06-26 10:32:22 +0000
Running preparation steps required to run the next scenarios
================================================================================
Make sure Foreman DB is up: 
/ Checking connection to the Foreman DB                               [OK]      
--------------------------------------------------------------------------------
Make sure Candlepin DB is up: 
- Checking connection to the Candlepin DB                             [OK]      
--------------------------------------------------------------------------------
Make sure Pulpcore DB is up: 
\ Checking connection to the Pulpcore DB                              [OK]      
--------------------------------------------------------------------------------


Running Backup
================================================================================
Check if the incremental backup has the right type:                   [FAIL]
The existing backup has PostgreSQL as a tarball, but the new one will have a dump.
--------------------------------------------------------------------------------
Scenario [Backup] failed.

The following steps ended up in failing state:

  [backup-incremental-parent-type]

Resolve the failed steps and rerun the command.
In case the failures are false positives, use
--whitelist="backup-incremental-parent-type"



Running Failed backup cleanup
================================================================================
Start applicable services: 

Starting the following service(s):
redis, postgresql, pulpcore-api, pulpcore-content, pulpcore-worker@1.service, pulpcore-worker@2.service, tomcat, dynflow-sidekiq@orchestrator, foreman, httpd, dynflow-sidekiq@worker-1, dynflow-sidekiq@worker-hosts-queue-1, foreman-proxy
- All services started                                                [OK]      
--------------------------------------------------------------------------------
Clean up backup directory:                                            [OK]
--------------------------------------------------------------------------------

Done with backup: 2024-06-26 10:32:23 +0000
Backup didn't finish. Incomplete backup was removed.
```

Gotta find out how to move that check to the prep steps.

**Review by ehelms — APPROVED** [review]

**Comment by ehelms** [comment]
@evgeni Are you holding on merge to perform testing first?

**Comment by evgeni** [comment]
Yeah, we were trying to run tests and they ended up red, so … waiting

**Comment by evgeni** [comment]
ah, nice. aaac56d5f27557a2257b3cb93224c5969e08f9d9 did actually break things for this PR \o/

**Review by ehelms — APPROVED** [review]

**Pull request merged**